src/main/dynamo/util.ts

```diff
@@ -10,26 +10,34 @@ export function toJSON<T>(item: { [key: string]: any }): T {
 
 export async function encryptOrPassThroughItems<T extends Record<string, any>>(
   jayz: JayZ | undefined,
-  item: T,
+  items: T[],
   encryptionBlacklist: Set<string>
-): Promise<MaybeEncryptedItems<T>> {
+): Promise<MaybeEncryptedItems<T>[]> {
   if (jayz !== undefined) {
-    const fieldsToEncrypt = Object.keys(item).filter(
-      (_) => !encryptionBlacklist.has(_)
-    )
-    return jayz.encryptItem(item, fieldsToEncrypt)
+    const itemsToEncrypt = items.map((item) => {
```

Comment:
This loop is ripe for optimization. And the way we could do that (in a future PR) is to have our code generator calculate this in advance (per model)

Reply:
I was wanted to see both upgrade instead of just one.